### ISSUE TITLE
Configure swipe up gesture to open search instead

### ISSUE THREAD

**Issue by KorigamiK** [MM2-0/Kvaesitso#1164]
Right now there's no way to change the place where widgets are placed. 
I need a way to swipe up to open search, and swipe left/right/down for seeing the widgets instead.

**Comment by neingeist** [comment]
I came here to request this as well :) I'm used to swipe up to open an "app drawer" and it's a bit irritating that here it's the other way!

But all in all I'm liking Kvaesitso very much, Thanks!

**Comment by KorigamiK** [comment]
@MM2-0 I would love to hear about the feasibility of this, this would be a perfect addition for the launcher I would want.

I'd love to contribute so feel free to explain how we would go about dong this :)

**Comment by MM2-0** [comment]
> @MM2-0 I would love to hear about the feasibility of this, this would be a perfect addition for the launcher I would want.
> 
> I'd love to contribute so feel free to explain how we would go about dong this :)

Pretty difficult. It would require a rewrite of major parts of the UI. Currently, there are two different scaffolds, pull down and pager. This is already pretty annoying because there is a lot of duplicate code. So instead of introducing a third scaffold, we should rewrite the entire thing and try to combine them into one single scaffold. This would also allow us to add a "no gesture" option.

Then there is the question what to do with the widgets? I guess the most sensible answer would be to keep the clock widget on the main screen and move all other widgets on a secondary page?

**Comment by KorigamiK** [comment]
Yes I am able to see that the code for animation and gestures is very coupled. 

I'd suggest having the widgets in another scaffold is alright and doesn't seem like a problem if I can scroll left or right to reveal them. 

The search bar may expand to the search scaffold on gesture up. 

Does it sound doable?

**Comment by KorigamiK** [comment]
And having the ability to disable widgets entirely would be cool as well!

**Comment by neingeist** [comment]
> Then there is the question what to do with the widgets? I guess the most sensible answer would be to keep the clock widget on the main screen and move all other widgets on a secondary page?

I've been using Kvaesitso for about a day now, so I'm probably a bit ignorant. I really like how the widgets are done here, I just wish I could get to them by swiping down/scrolling up (and so having swipe up free to get to the "app drawer")

**Comment by zakir786lrk** [comment]
I came to request the similar changes:

1. There should be function to add multiple (2 or 3) rows of favourites on homescreen which would be accessible without swipe gesture. 

2. Instead of "search/app drawer" gesture setting there should be swipe up and swipe down gestures where we can set swipe down as open notification/quick settings. And swipe up to open search/app drawer. There should be long press to edit homescreen (clock widget, favorites, wallpaper, etc).

3. swipe left and right can be set as  (widgets screen), (news feed) or (notifications) etc. 

4. There can be a native widget containing favourite quick setting shortcuts like wifi, bluetooth etc

**Comment by and1mon** [comment]
> And having the ability to disable widgets entirely would be cool as well!

Yes, this would be awesome. Even when removing all widgets, i can still swipe up just a tiny bit, which is super annoying when done by accident.

**Comment by deathblade666** [comment]
Just wanted to chime in, and say I also would like to see this as an option. I do see how it would be quite a large change given how its currently built out and as @MM2-0 said there is the question of how to handle widgets. 

Regarding widgets, one potentially possible avenue would be using a bottom sheet instead of say a right or left page or similar. Though, I'm not sure if there are limitations that would not allow it to work or not. But if interested I have a proof of concept app (written in flutter) in my repos.

**Comment by OlliesGudh** [comment]
I absolutely support this idea!
Not only that, I just like using the swipe down for quick settings and as long as the app drawer is locked onto this gesture it will be impossible to assign the gesture to anything else.
I just migrated from [KISS launcher](https://github.com/Neamar/KISS) because I was experiencing some issues regarding widgets auto-start after boot. It worked for me regarding this issue, but Kvaesitso's UX regarding gestures really grinds my gears, sideways swipes aren't all that smooth at least in my device comparing to KISS'.
Please note that I LOVED Kvaesitso and will keep using it. Would def volunteer if I could code. Props to the devs nonetheless! Killer launcher!

**Comment by neingeist** [comment]
Thanks for making this possible :)

**Comment by KorigamiK** [comment]
This is absolutely perfecto dude! You're the GOAT @MM2-0 !!
